addons/account/models/account_account.py

```diff
@@ -1104,8 +1104,17 @@ def _load_records_write(self, values):
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_contains_journal_items(self):
-        if self.env['account.move.line'].search_count([('account_id', 'in', self.ids)], limit=1):
-            raise UserError(_('You cannot perform this action on an account that contains journal items.'))
+        accounts_with_items = [
+            account['account_id'][1]
+            for account in self.env['account.move.line'].sudo().read_group(
```

Comment:
Use `_read_group` if it's not a result to give to the (web)client

But now instead of doing an index scan (hopefully stopping early with the limit and the company filter), you are forcing a full seq scan of the table